test/e2e/test/elasticsearch/checks_http.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elasticsearch
+
+import (
+	"context"
+	"crypto/x509"
+
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/reconcile"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/services"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/sset"
+	esuser "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/user"
+	"github.com/elastic/cloud-on-k8s/pkg/dev/portforward"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/net"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+)
+
+func CheckHTTPConnectivityWithCA(es esv1.Elasticsearch, k *test.K8sClient, caCert []*x509.Certificate) error {
+	password, err := k.GetElasticPassword(k8s.ExtractNamespacedName(&es))
+	if err != nil {
+		return err
+	}
+	user := client.BasicAuth{Name: esuser.ElasticUserName, Password: password}
+
+	pods, err := sset.GetActualPodsForCluster(k.Client, es)
+	if err != nil {
+		return err
+	}
+	var dialer net.Dialer
+	if test.Ctx().AutoPortForwarding {
+		dialer = portforward.NewForwardingDialer()
+	}
+	v, err := version.Parse(es.Spec.Version)
+	if err != nil {
+		return err
+	}
+
+	for _, p := range reconcile.AvailableElasticsearchNodes(pods) {
```

Comment:
While re-reading this check I was wondering if we could "miss" some Pods, either because some of them are being recreated or because they are not ready.
I guess that if we do not want to make any progress until all the expected Pods are exposing the desired CA the algorithm would be:

1. Get the actual nodes, if at least one is not available/ready return an error
2. Get all the expected nodes, if the list does not match the actual one return an error
3. For each node check the connectivity using the desired CA

This situation is very unlikely to happen though, so I'm not sure if it's worth to improve this check.

Reply:
I think you are right that is unlikely. All upgrades in this test are not leading to actual Pod rotations and we check the readiness of the expected Pods during the initial creation. So I am tempted to leave it as is for now.